### PR TITLE
[Doc] clean useless files

### PR DESCRIPTION
**Pull request by J-shang**
### Description ###
~TODO: rename python_api_ref to python_api~

### CODE REVIEW

File: docs/tools/md2rst.py [hunk 1]
```diff
@@ -82,7 +82,7 @@ def process_github_link(line):
 for root, dirs, files in os.walk('en_US'):
     root = Path(root)
     for file in files:
-        if not file.endswith('.md') or file == 'Release_v1.0.md':
+        if not file.endswith('.md'):
```
Comment (matluster) [inline docs/tools/md2rst.py:85]: No need to update this file.
Actually `tools/md2rst.py` and `tools/patches` are no longer needed. Feel free to delete them.

File: docs/source/notes/architecture_overview.rst [hunk 2]
```diff
@@ -75,11 +75,11 @@ NNI provides an easy-to-use model compression framework to compress deep neural
...
 
-A detailed description of model compression and its usage can be found `here <Compression/Overview.rst>`__.
+A detailed description of model compression and its usage can be found `here <../compression/index.rst>`__.
 
 Automatic Feature Engineering
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Automatic feature engineering is for users to find the best features for their tasks. A detailed description of automatic feature engineering and its usage can be found `here <FeatureEngineering/Overview.rst>`__. It is supported through NNI trial SDK, which means you do not have to create an NNI experiment. Instead, simply import a built-in auto-feature-engineering algorithm in your trial code and directly run your trial code. 
+Automatic feature engineering is for users to find the best features for their tasks. A detailed description of automatic feature engineering and its usage can be found `here <../feature_engineering/overview.rst>`__. It is supported through NNI trial SDK, which means you do not have to create an NNI experiment. Instead, simply import a built-in auto-feature-engineering algorithm in your trial code and directly run your trial code.
```
Comment (matluster) [inline docs/source/notes/architecture_overview.rst:83]: Use `:doc:` please.

File: docs/source/notes/architecture_overview.rst [hunk 3]
```diff
@@ -62,7 +62,7 @@ This is a core and basic feature of NNI, we provide many popular `automatic tuni
 General NAS Framework
 ^^^^^^^^^^^^^^^^^^^^^
 
-This NAS framework is for users to easily specify candidate neural architectures, for example, one can specify multiple candidate operations (e.g., separable conv, dilated conv) for a single layer, and specify possible skip connections. NNI will find the best candidate automatically. On the other hand, the NAS framework provides a simple interface for another type of user (e.g., NAS algorithm researchers) to implement new NAS algorithms. A detailed description of NAS and its usage can be found `here <NAS/Overview.rst>`__.
+This NAS framework is for users to easily specify candidate neural architectures, for example, one can specify multiple candidate operations (e.g., separable conv, dilated conv) for a single layer, and specify possible skip connections. NNI will find the best candidate automatically. On the other hand, the NAS framework provides a simple interface for another type of user (e.g., NAS algorithm researchers) to implement new NAS algorithms. A detailed description of NAS and its usage can be found `here <../nas/index.rst>`__.
```
Comment (matluster) [inline docs/source/notes/architecture_overview.rst:65]: Same here.

File: docs/source/notes/architecture_overview.rst [hunk 4]
```diff
@@ -62,7 +62,7 @@ This is a core and basic feature of NNI, we provide many popular `automatic tuni
 General NAS Framework
```
Comment (matluster) [inline docs/source/notes/architecture_overview.rst:62]: Did we talk about merging `notes` and `misc`?

Comment (J-shang) [reply docs/source/notes/architecture_overview.rst:62]: yes, I will merge `notes` and `misc` to `misc`

File: docs/tools/md2rst.py (hunk 1 above)
Comment (J-shang) [reply docs/tools/md2rst.py]: removed

File: docs/source/notes/architecture_overview.rst (hunk 2 above)
Comment (J-shang) [reply docs/source/notes/architecture_overview.rst]: fixed

File: docs/source/notes/architecture_overview.rst (hunk 3 above)
Comment (J-shang) [reply docs/source/notes/architecture_overview.rst]: fixed

**Review by ultmaster — APPROVED** [review]

**Review by liuzhe-lz — APPROVED** [review]

**Pull request merged**